### PR TITLE
feat: check if batch is already responded

### LINKED ISSUE #172: feat: check if task is already responded in ServiceManager
# Description
In respondToTask method check if the task has already been responded

### PR DESCRIPTION
**Pull request by JuArce**
respondToTask method now checks if batch was already responded

To test you can duplicate lines 200-203 from `aggregator/internal/pkg/aggregator.go`
``` go
	_, err := agg.avsWriter.SendAggregatedResponse(context.Background(), batchMerkleRoot, nonSignerStakesAndSignature)
	if err != nil {
		agg.logger.Error("Aggregator failed to respond to task", "err", err)
	}
```
So the second time it tries to respond the task, the service manager will revert the transaction

### CODE REVIEW

**Review by taturosati — APPROVED** [review]
lgtm

File: contracts/src/core/AlignedLayerServiceManager.sol [hunk 1]
```diff
@@ -124,6 +128,8 @@ contract AlignedLayerServiceManager is ServiceManagerBase, BLSSignatureChecker {
             "Signatories do not own at least threshold percentage of a quorum"
         );
 
+        batchesState[batchMerkleRoot].responded = true;
```
Comment (uri-99) [inline contracts/src/core/AlignedLayerServiceManager.sol]: This assignment should be right after the `require` statement it is involved in (line 104), to follow CEI pattern and thus avoid possible re-entrancy attacks.
Even though this function has not much value to be exploited with a reentrancy, it is always a good idea to follow precaution and safety patterns 👍

**Review by uri-99 — CHANGES_REQUESTED** [review]
👀

**Review by taturosati — APPROVED** [review]

**Review by uri-99 — APPROVED** [review]
👍

**Pull request merged**